src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/WindowsForms/MyAppFileAccessor.cs

```diff
@@ -65,43 +68,35 @@ private void DocData_Modifying(object sender, EventArgs e)
 
     private async Task<MyAppDocument?> TryGetMyAppFileAsync()
     {
-        if (_docData?.Data is null || !File.Exists(_absolutePath))
+        if (_docData?.Data is null)
         {
-            await _threadingService.SwitchToUIThread();
-            try
-            {
-                _docData = new DocData(_serviceProvider, _absolutePath);
-                _docData.Modifying += DocData_Modifying;
-            } 
-            catch (Exception)
+            await TaskScheduler.Default;
+            // Create My Project directory if it doesn't exist. If it does, nothing happens
+            await _storage.CreateFolderAsync("My Project");
+
+            if (!File.Exists(_absolutePath))
             {
-                // If we have reached here, it means that the file doesn't exist in that location, so try to create it
-                
-                // Create My Project directory if it doesn't exist. If it does, nothing happens
-                await _storage.CreateFolderAsync("My Project");
-                
                 // Create and write defaults to the myapp file
                 await _storage.CreateEmptyFileAsync(_absolutePath);
-                StreamWriter writer = File.AppendText(_absolutePath);
+                using StreamWriter writer = File.AppendText(_absolutePath);
                 await writer.WriteAsync(DefaultMyappFileContents);
                 await writer.FlushAsync();
-                writer.Close();
-                
-                // Now that we've created the file, we can instantiate _docData and myAppDocument.
-                _docData = new DocData(_serviceProvider, _absolutePath);
-                _docData.Modifying += DocData_Modifying;
-
-                myAppDocument = new MyAppDocument(_docData);
-                return myAppDocument;
             }
-            
-            await TaskScheduler.Default;
         }
-
-        if (myAppDocument is null)
+        
+        await _threadingService.SwitchToUIThread();
+        try
         {
+            _docData = new DocData(_serviceProvider, _absolutePath);
+            _docData.Modifying += DocData_Modifying;
             myAppDocument = new MyAppDocument(_docData);
         }
+        catch (Exception)
+        {
+            // If we've reached here, the file write succeeded but the file may have been deleted by the time UI thread is switched to
+        }
+
+        await TaskScheduler.Default;
```

Comment:
I don't think switching back here does anything useful. What happens in the caller if you remove this?

Reply:
I thought this was your prior recommendation? All seems good if I remove this

Reply:
> I thought this was your prior recommendation?

My suggestion was to use a local function to enclose all the stuff that needs the UI thread. Something like this:

```c#
// on background thread
await LocalFunctionAsync();
// still on background thread

async Task LocalFunctionAsync()
{
    await SwitchToMainThreadAsync();
    // on main thread
}
```

In your case, you don't do any more real work after the main thread code, so it's pretty much equivalent.